python/pyspark/cloudpickle.py

```diff
@@ -801,10 +801,10 @@ def save_ellipsis(self, obj):
     def save_not_implemented(self, obj):
         self.save_reduce(_gen_not_implemented, ())
 
-    if PY3:
-        dispatch[io.TextIOWrapper] = save_file
-    else:
+    try:               # Python 2
         dispatch[file] = save_file
+    except NameError:  # Python 3
+        dispatch[io.TextIOWrapper] = save_file
```

Comment:
is it our preference to check for python version or handle error?

Reply:
Actually not quite sure if we should do the same thing in our PySpark codebase .. but it's the decision made in cloudpickle .. it's virtually the same anyway